lib/segment/src/vector_storage/simple_sparse_vector_storage.rs

```diff
@@ -152,8 +153,12 @@ impl VectorStorage for SimpleSparseVectorStorage {
 
     fn insert_vector(&mut self, key: PointOffsetType, vector: VectorRef) -> OperationResult<()> {
         let vector: &SparseVector = vector.try_into()?;
-        debug_assert!(vector.is_sorted());
```

Comment:
Being sorted is not part of the validation, it is an expectation of the persistence layer.
Please reintroduce `debug_assert!(vector.is_sorted());` :)

Reply:
Placed it back, thanks for clarifying =)